observer.go

```diff
@@ -39,50 +43,27 @@ type observer struct {
 	observers []Observer
 }
```

Comment:
Is this still needed? I thought `compositeObserver` above would be used instead,